pkg/puller/downloader.go

```diff
@@ -0,0 +1,173 @@
+package puller
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"hash/fnv"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+var (
+	d *Downloader
+)
+
+type Downloader struct {
+	NumRetries int
+	ModelDir   string
+}
+
+type Protocol string
+
+const (
+	// Supported Protocols
+	S3    Protocol = "s3://"
+	GS    Protocol = "gs://"
+	PVC   Protocol = "pvc://"
+	File  Protocol = "file://"
+	HTTPS Protocol = "https://"
+)
+
+// TODO: Add more supported protocols
+var SupportedProtocols = []Protocol{S3}
+
+func init() {
+	d = NewDownloader()
+}
+
+func NewDownloader() *Downloader {
+	d := new(Downloader)
+	return d
+}
+
+func SetEnvs(numRetries int, modelDir string) {
```

Comment:
This function name indicate that it is setting an environment variable. Can we rename it to something less ambiguous ?